src/mpi/mpi_transceiver.h

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+* Copyright 2014-2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef _MPI_TRANSCEIVER_INCLUDED_
+#define _MPI_TRANSCEIVER_INCLUDED_
+
+#include "transceiver.h"
+
+// implementation of transceiver_iface using MPI
+class mpi_transceiver : public transceiver_impl
+{
+public:
+    mpi_transceiver() : transceiver_impl() {}
+
+    virtual void init();
+
+    virtual void fini();
+    
+    virtual size_t nMembers();
+
+    virtual size_t me();
```

Comment:
it would be better to rename `me` to `id`

Reply:
Not sure I agree. id of what?
Any other opinion?

> It is not clear why so complicated inheritance schema used.

What is unclear? I hoped the comments clarify the intent. I see no other way to provide the same functionality with a clean separation of concerns. Suggestions?